ci-operator/step-registry/baremetalds/sno/rename/baremetalds-sno-rename-commands.sh

```diff
@@ -0,0 +1,317 @@
+set -o errexit
+set -o pipefail
+
+# Fetch packet basic configuration
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/packet-conf.sh"
+
+cat >"${SHARED_DIR}"/recert-rename-sno.sh << "EOF"
+#!/usr/bin/env bash
+
+export PREVIOUS_CLUSTER_NAME="${PREVIOUS_CLUSTER_NAME:-test-infra-cluster}"
+export PREVIOUS_BASE_DOMAIN="${PREVIOUS_BASE_DOMAIN:-redhat.com}"
+export CLUSTER_NAME="${CLUSTER_NAME:-another-name}"
+export BASE_DOMAIN="${BASE_DOMAIN:-another.domain}"
+export SINGLE_NODE_IP="${SINGLE_NODE_IP:-192.168.127.10}"
+
+# assisted-test-infra sets up 2 network interfaces that compete with each other
+# when setting the NODE_IP and KUBELET_NODEIP. Use KUBELET_NODEIP_HINT to
+# ensure the correct interface is chosen.
+#
+# https://access.redhat.com/articles/6956852
+ssh -o UserKnownHostsFile=/dev/null \
+  -o StrictHostKeyChecking=no \
+  -o LogLevel=ERROR \
+  "core@${SINGLE_NODE_IP}" \
+  "echo 'KUBELET_NODEIP_HINT=192.168.127.1' | sudo tee /etc/default/nodeip-configuration"
+
+VMS=( $( virsh list --all --name ) )
+for vm in ${VMS[@]}; do
+  virsh reboot ${vm}
+  until virsh domstate ${vm} | grep "running"; do
+    echo "${vm} not yet running"
+    sleep 5
+  done
+done
+
+echo "Waiting for API before applying the rename and dnsmasq MachineConfigs..."
+until oc get clusterversion &> /dev/null
+do
+  echo "Waiting for API before applying the rename and dnsmasq MachineConfigs..."
+  sleep 5
+done
+echo "API is available"
+
+rename_script=$(cat <<IEOF
+#!/usr/bin/env bash
+
+set -euoE pipefail
+
+export KUBECONFIG=/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/localhost.kubeconfig
+function wait_for_api {
+  echo "Waiting for API..."
+  until oc get clusterversion &> /dev/null
+  do
+    echo "Waiting for API..."
+    sleep 5
+  done
+  echo "API is available"
+}
+
+function start_containers {
+  systemctl start crio
+  systemctl start kubelet
+}
+
+function stop_containers {
+  systemctl stop kubelet
+  crictl ps -q | xargs crictl stop || true
+  crictl rmp -a -f || true
+  systemctl stop crio
+}
+
+function recert {
+  local release_image="${RELEASE_IMAGE:-quay.io/openshift-release-dev/ocp-release:4.14.2-x86_64}"
+  local etcd_image="\$(oc adm release extract --from="\${release_image}" --file=image-references | jq '.spec.tags[] | select(.name == "etcd").from.name' -r)"
+  local recert_image="${RECERT_IMAGE:-quay.io/edge-infrastructure/recert:latest}"
+  local previous_base_domain="${PREVIOUS_BASE_DOMAIN:-redhat.com}"
+  local previous_cluster_name="${PREVIOUS_CLUSTER_NAME:-test-infra-cluster}"
+  local base_domain="${BASE_DOMAIN:-another.domain}"
+  local cluster_name="${CLUSTER_NAME:-another-name}"
+
+  podman run --authfile=/var/lib/kubelet/config.json \
+      --name recert_etcd \
+      --detach \
+      --rm \
+      --network=host \
+      --privileged \
+      --entrypoint etcd \
+      -v /var/lib/etcd:/store \
+      "\${etcd_image}" \
+      --name editor \
+      --data-dir /store \
+
+  sleep 10
```

Comment:
Can we properly check the etcd health endpoint instead?

Reply:
Yes, I hate sleep as well, I'll issue a request to the etcd API with `curl` probably (since we don't have `etcdctl` in there) :+1: